lab1/lab1.rst

```diff
@@ -0,0 +1,412 @@
+.. raw:: latex
+
+   \LARGE
+
+**ECS 154B Lab 1, Spring 2017**
+
+**Due by 11:59 PM on April 12, 2017**
+
+**Via Canvas**
+
+Goals
+=====
+
+-  Learn how to use Logisim.
+
+-  Learn how to use the circuit analyzer tool.
+
+-  Learn an alternative to combinational logic.
+
+Logisim
+=======
+
+-  The project page for Logisim is on `Dr. Carl Burch’s
+   website <http://www.cburch.com/logisim/>`__.
+
+-  You can download Logisim on
+   `SourceForge <http://sourceforge.net/projects/circuit/>`__ for
+   Windows and OS X. If you are using Linux, your package manager for
+   your distribution may have Logisim.
+
+-  You can find a brief tutorial for Logisim on `Professor Farrens’
+   class
+   website <http://american.cs.ucdavis.edu/academic/ecs154a/postscript/logisim-tutorial.pdf>`__.
+
+-  The **User’s Guide** and **Library Reference** in the Help menu in
+   Logisim are also very helpful.
+
+Introduction
+============
+
+When designing a circuit to implement a function, there are two
+different design routes that you can take: use pure combinational logic,
+or use a Read Only Memory (ROM). Combinational logic involves combining
+AND, OR, and NOT gates to implement the Boolean equation that you derive
+from the truth table. On the other hand, by using a ROM, you simply
+implement the truth table in hardware.
+
+ROM Implementation
+==================
+
+A memory unit can be viewed as simply a truth table in hardware. Here
+are the steps to creating a ROM implementation of a truth table.
+
+#. Create a ROM, where the number of addressing bits is equal to the
+   number of inputs, and the data bit width is equal to the number of
+   output bits.
+
+#. Using the input bits as the address, fill in the entries of the ROM
+   with the correct outputs for that combination of inputs.
+
+#. To get the output, address the ROM using the concatenation of the
+   input signals.
+
+Example
+-------
+
+The following is an example of implementing the XOR function using
+microcode. Here is the truth table for XOR
+:math:`\left( \oplus \right)`.
+
++---+---+--------------------+
+| X | Y | X :math:`\oplus` Y |
++---+---+--------------------+
+| 0 | 0 | 0                  |
++---+---+--------------------+
+| 0 | 1 | 1                  |
++---+---+--------------------+
+| 1 | 0 | 1                  |
++---+---+--------------------+
+| 1 | 1 | 0                  |
++---+---+--------------------+
+
+Applying step 1, we first create a ROM with 2 address bits, because we
+have 2 inputs. Each entry is 1 bit wide, because there is only output.
+Next, we fill in the ROM using X and Y as the address bits.
+
++---+---+----------------------------------------+-------+
+| X | Y | Address                                | Value |
++===+===+========================================+=======+
+| 0 | 0 | :math:`00 \left( \mathtt{0x0} \right)` | 0     |
```

Comment:
This, I think you can use ``0x0`` for inline verbatim. Though, this might not work inside a table, which is OK if it doesn't work.